### PR TITLE
Use a single client instance during a transaction

### PR DESCRIPTION
**Pull request by grahambates**
Possible fix for #29

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -92,7 +92,7 @@ module.exports = Adapter => class PostgreSQLAdapter extends Adapter {
       this.pool = new Pool(connection)
     }
 
-    // Set `client` alias for backwards compatibility.
+    // This is overridden with a single client instance during a transaction.
     this.client = this.pool
```
Comment (daliwali) [inline lib/index.js]: I'm not sure if any of the changes below affect anything, since `this.client` is effectively an alias for `this.pool`.

Comment (grahambates) [reply lib/index.js]: Although initially it's an alias for `this.pool`, during a transaction it's overridden with a client instance. See `lib/index.js:562` where it's replaced in the scope with `Object.assign()`.

Now that the CRUD methods are directly using `this.client` it's no longer just for backwards compatibility.

**Comment by Voles** [comment]
@grahambates does this fix resolve the issue in the [example app](https://github.com/grahambates/fortune-postgres-example) you've setup?

**Comment by Voles** [comment]
I've tested the fix included in this PR on one of my applications and the deadlock seems to be gone.

**Comment by grahambates** [comment]
@Voles Yes it fixes the issue for me.

**Comment by daliwali** [comment]
Sorry about the delay, I guess this does solve the problem. Will release a patch version.

**Comment by Voles** [comment]
@daliwali no need to rush at all. Thanks for taking the time helping us with this! 🌷

**Pull request merged**